example/lib/navigation/navigator_bloc.dart

```diff
@@ -0,0 +1,26 @@
+import 'dart:async';
+import 'package:blemulator_example/peripheral_details/peripheral_details_screen.dart';
+import 'package:bloc/bloc.dart';
+import 'package:flutter/material.dart';
+import './bloc.dart';
+
+class NavigatorBloc extends Bloc<NavigatorEvent, dynamic> {
+  final GlobalKey<NavigatorState> navigatorKey;
+
+  NavigatorBloc({this.navigatorKey});
+
+  @override
+  dynamic get initialState => 0;
+
+  @override
+  Stream<dynamic> mapEventToState(
+    NavigatorEvent event,
+  ) async* {
+    if (event is NavigatorPop) {
```

Comment:
does the method generate any stream?

Reply:
It does not, but after some discussions we decided to leave it as is now to use `flutter_bloc`s `BlocProvider` capabilities for providing `NavigatorBloc` down the widget tree. Separate issue was added to analyze if it is worth writing dedicated component for that (`NavigatorProvided`?)